component/component.go

```diff
@@ -68,17 +66,17 @@ type Factory interface {
 // the configuration loading system will use to unmarshal the config.
 type ConfigUnmarshaler interface {
 	// Unmarshal is a function that un-marshals a viper data into a config struct in a custom way.
-	// componentViperSection *viper.Viper
+	// componentSection map[string]interface{}
 	//   The config for this specific component. May be nil or empty if no config available.
 	// intoCfg interface{}
 	//   An empty interface wrapping a pointer to the config struct to unmarshal into.
-	Unmarshal(componentViperSection *viper.Viper, intoCfg interface{}) error
+	Unmarshal(componentSection map[string]interface{}, intoCfg interface{}) error
```

Comment:
What do you think, should we expose maybe a struct `ConfigLoader` (or a better name) where we can start adding functionalities like `UnmarshalExact`/`ToStringMap`. Then users will no longer need to use something like viper (we will use to implement these funcs and fix all the bugs). Then we can kind of control all the implementations and if a bug is discovered fix all of them?

Right now we can just have:

```
type ConfigLoader struct {
  v *viper.Viper
}

func (cl *ConfigLoader) UnmarshalExact(interface) error {
}

func (cl *ConfigLoader) ToStringMap(interface) map[string]interface{} {
}
```

Not sure if this is the right thing, but I want a second opinion from someone who started to work on this :)

Reply:
I am not sure I understand where `ConfigLoader` would be used. Would it be on the `CustomUnmarshaler` interface, as in
```go
 type CustomUnmarshaler func(componentSection map[string]interface{}, intoCfg interface{}) error
```
?

Would it be passed to the `config.Load` function? Or something else?

Reply:
🤦 sorry I meant to say that on the code haha. I think that makes sense and it's better than the `map[string]interface{}` thing, I will change it (and see if I can come up with a better name)